Source/WebCore/rendering/shapes/Shape.cpp

```diff
@@ -102,7 +102,7 @@ Ref<const Shape> Shape::createShape(const BasicShape& basicShape, const LayoutPo
     switch (basicShape.type()) {
 
     case BasicShape::Type::Circle: {
-        const auto& circle = downcast<BasicShapeCircle>(basicShape);
+        const auto& circle = uncheckedDowncast<BasicShapeCircle>(basicShape);
```

Comment:
Out of curiosity: do these unchecked casts give any perf improvement?

Reply:
They avoid a redundant type check. I am still trying to make the release type check affordable on Intel so I can enable it there.